### ISSUE TITLE
Upgrade to Node v8.6.0 (N-API)

### ISSUE THREAD

**Issue by omichelsen** [electron/electron#10629]
Version 8.6.0 removes the experimental flag for N-API support which was introduced in Node v8.0.0.

We would very much like to be able to use this API for Electron apps.

A few references to N-API docs:
[Node.js v8.6.0 Documentation](https://nodejs.org/api/n-api.html)
[N-API: Next generation Node.js APIs for native modules](https://medium.com/the-node-js-collection/n-api-next-generation-node-js-apis-for-native-modules-169af5235b06)

**Comment by codebytere** [comment]
We upgrade node periodically, and i believe have passed this version now. In this event, closing this issue.

**Comment by farism** [comment]
@codebytere I'm not sure if Electron is using a version of Node that has removed the experimental flag. The following are with `electron@1.8.2`:

`process.versions.node` is returning `8.2.1`

<img width="497" alt="screen shot 2018-02-11 at 4 50 44 pm" src="https://user-images.githubusercontent.com/3803714/36080866-f2184b2e-0f4b-11e8-9bc2-e2090e909ef5.png">

Trying to use a N-API module is giving me this error

```
~/Documents/Projects/autohex/node_modules/bindings/bindings.js:88 Uncaught Error: The module '~/Documents/Projects/autohex/packages/io/build/Release/hello.node'
was compiled against the ABI-stable Node.js API (N-API).
This feature is experimental and must be enabled on the 
command-line by adding --napi-modules.
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:172:20)
    at Object.Module._extensions..node (module.js:598:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:172:20)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at bindings (~/Documents/Projects/autohex/node_modules/bindings/bindings.js:81:44)
    at Object.<anonymous> (~/Documents/Projects/autohex/packages/io/src/main.js:1:187)
```

Trying to pass in `--napi-modules` flag through the `--js-flags` is unrecognized

```
electron --js-flags="--napi-modules" src/main.js

Error: unrecognized flag --napi-modules
```

Without actually upgrading to Node >= 8.6.0, what is the proper way to pass this flag in through Electron?

**Comment by codebytere** [comment]
oh sorry you're correct, the 1-8-x line is using `v8.2.1`. The next released line will likely use `v8.9.3`.

**Comment by farism** [comment]
Cool, sounds like it will be coming soon then. Thank you for the update.

Is there currently a way to pass this flag in?
